python/packages/nisar/workflows/stage_watermask.py

```diff
@@ -382,43 +382,45 @@ def download_watermask(polys, epsgs, outfile, version):
             xmin, ymin, xmax, ymax = poly.bounds
             translate_watermask(vrt_filename, outpath, xmin, xmax, ymin, ymax)
 
-        # Get the WATER description from the README.txt file using GDAL
-        in_readme_path = vrt_filename.replace(f'EPSG{epsg}.vrt', f'EPSG{epsg}/LICENSE.txt')
-        try:
-            water_descr = parse_readme_fields(in_readme_path)
-
-            # Build vrt with downloaded watermasks
-            vrt_dataset = gdal.BuildVRT(outfile, watermask_list)
-            # vrt_dataset.SetMetadataItem("dem_description", f'{dem_descr}')
-            for key, val in water_descr.items():
-                # # convert the bullet header to a tidy metadata key
-                # clean_key = key.lower().replace(" ", "_")
-                # vrt_dataset.SetMetadataItem(clean_key, val)
-                vrt_dataset.SetMetadataItem(key, val)
-        except Exception as e:
+        # Build vrt with downloaded watermasks
+        vrt_dataset = gdal.BuildVRT(outfile, watermask_list)
+
+        # Get the WATER description from the LICENSE.txt file using GDAL
+        in_license_path = vrt_filename.replace(f'EPSG{epsg}.vrt', f'EPSG{epsg}/LICENSE.txt')
+        license_exists = gdal.VSIStatL(in_license_path) is not None
+
+        if license_exists:
+            try:
+                water_descr = parse_license_fields(in_license_path)
+                for key, val in water_descr.items():
+                    vrt_dataset.SetMetadataItem(key, val)
+
+            except Exception as e:
+                warnings.warn(
+                    f"Found {license_path} but could not parse it ({e}). "
+                    "Proceeding without metadata."
+                )
+        else:
             warnings.warn(
-                f"Could not read or parse README.txt at {in_readme_path} "
-                f"({e}). Proceeding without metadata."
+                f"No LICENSE.txt found at {in_license_path} "
+                f"(expected for WATERMASK v{version}). Proceeding without metadata."
             )
-
     except Exception:
```

Comment:
```suggestion
    except Exception as e:
```

Needed for this line below:

https://github.com/isce-framework/isce3/blob/60d9072af860638651365bd47974d5fad8b40166/python/packages/nisar/workflows/stage_watermask.py#L411